### PR TITLE
Add addKvs, addEsdts, and addMappers methods to FSWorld and LSWorld

### LINKED ISSUE #200: Add a method to easily add Kvs to an existing account
To be able to add Kvs to an existing account, we need to do something like this:
```ts
await contract.setAccount({
  ...(await contract.getAccountWithoutKvs()),
  kvs: [
    await contract.getAccountKvs(),
    e.kvs
      .Mapper("points_by_user_by_pool", e.Usize(1), e.U8(0))
      .Value(e.U(10)),
    e.kvs
      .Mapper("points_by_user_by_pool", e.Usize(1), e.U8(1))
      .Value(e.U(100)),
  ],
});
```
which is not very readable nor intuitive.

Or create this kind of helper function:
```ts
export const giveTokensTo = async (wallet: FSWallet, tokens: Esdt[]) => {
  await wallet.setAccount({
    ...(await wallet.getAccountWithoutKvs()),
    kvs: [await wallet.getAccountKvs(), e.kvs.Esdts(tokens)],
  });
};
```

We could add an `appendKvs` method to the `Proxy` and `World`classes:
```ts
appendKvs(address: AddressLike, kvs: EncodableKvs, options?: { overwrite?: boolean }): Promise<void>;
```
And to the `Contract` and `Wallet` classes:
```ts
appendKvs(kvs: EncodableKvs, options?: { overwrite?: boolean }): Promise<void>;
```

We can discuss about the `options` parameter. But I think it's important to let people choose if they want to overwrite the existing Kvs or not.
I think the default behavior should be to overwrite the existing Kvs.

We can also discuss about the naming: `appendKvs`, `appendAccountKvs`, `appendAccount`, ...

Example usage:
```ts
await contract.appendKvs([
  e.kvs
    .Mapper("points_by_user_by_pool", e.Usize(1), e.U8(0))
    .Value(e.U(10)),
  e.kvs
    .Mapper("points_by_user_by_pool", e.Usize(1), e.U8(1))
    .Value(e.U(100)),
]);

await contract.appendKvs({
  mappers: [
    {
      key: ["points_by_user_by_pool", e.Usize(1), e.U8(0)],
      value: e.U(10),
    },
    {
      key: ["points_by_user_by_pool", e.Usize(1), e.U8(1)],
      value: e.U(100),
    },
  ],
});

await wallet.appendKvs({
  esdts: [
    {
      id: "TOKEN-123456",
      amount: 10n ** 18n,
    },
  ],
});
```

### PR DESCRIPTION
**Pull request by nvergez**
Closes #200

### CODE REVIEW

**Review by lcswillems — CHANGES_REQUESTED** [review]

File: xsuite/src/world/fsworld.ts [hunk 1]
```diff
@@ -185,6 +203,27 @@ export class FSWorld extends World {
     return super.deployContract(tx).then((r) => this.addContractPostTx(r));
   }
 
+  async addKvs(address: AddressLike, kvs: EncodableKvs) {
```
Comment (lcswillems) [inline xsuite/src/world/fsworld.ts:206]: "async" not necessary

Let's add this rule in the eslint config in error mode:

https://eslint.org/docs/latest/rules/require-await

File: xsuite-lightsimulnet/src/handleAdmin.go [hunk 2]
```diff
@@ -158,6 +231,16 @@ type RawAccount struct {
 	Owner					*string
 }
 
+type UpdateRawAccount struct {
```
Comment (lcswillems) [inline xsuite-lightsimulnet/src/handleAdmin.go:234]: Let's make `Nonce` and `Balance` optional in type `RawAccount` and use a default of 0 and 0 in `setAccount`:

```
worldAccount := &worldmock.Account{
  Nonce:           0,
  Balance:         big.NewInt(0),
  BalanceDelta:    big.NewInt(0),
  DeveloperReward: big.NewInt(0),
  Storage:         map[string][]byte{},
  MockWorld: 			 e.scenexec.World,
}
```

And let's remove type `UpdateRawAccount`.

File: xsuite-lightsimulnet/src/handleAdmin.go [hunk 3]
```diff
@@ -54,6 +54,24 @@ func (e *Executor) HandleAdminSetAccounts(r *http.Request) (interface{}, error)
...
+	if err != nil {
+		return nil, err
+	}
+	for _, rawAccount := range rawAccounts {
+		err = e.updateAccount(rawAccount)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return map[string]interface{}{
```
Comment (lcswillems) [inline xsuite-lightsimulnet/src/handleAdmin.go:70]: Let's introduce the variable `jData`.

In `HandleAdminSetAccounts`, let's return `return jData, nil`, which is indeed better.

File: xsuite-lightsimulnet/src/handleAdmin.go [hunk 4]
```diff
@@ -148,6 +166,61 @@ func (e *Executor) setAccount(rawAccount RawAccount) error {
...
+				return err
+			}
+			worldAccount.Storage[string(_key)], err = hex.DecodeString(value)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	// Do we want to update the code if rawAccount.Code == ""
+	if rawAccount.Code != nil && *rawAccount.Code != "" {
```
Comment (lcswillems) [inline xsuite-lightsimulnet/src/handleAdmin.go:197]: In setAccount, when it is == "", what's the value? nil?

In updateAccount, we should then set the same value (nil maybe).

Same reasoning applies for the other questions.

File: xsuite-lightsimulnet/src/handleAdmin.go [hunk 5]
```diff
@@ -148,6 +166,61 @@ func (e *Executor) setAccount(rawAccount RawAccount) error {
...
+	if rawAccount.CodeMetadata != nil && *rawAccount.CodeMetadata != "" {
+		worldAccount.CodeMetadata, err = hex.DecodeString(*rawAccount.CodeMetadata)
+		if err != nil {
+			return err
+		}
+	// Do we want to overwrite in any case the code metadata of a non-smart contract?
+	} else if !worldAccount.IsSmartContract {
+		worldAccount.CodeMetadata = (&vmcommon.CodeMetadata{ Readable: true }).ToBytes();
+	}
+	if rawAccount.Owner != nil && *rawAccount.Owner != "" {
```
Comment (lcswillems) [inline xsuite-lightsimulnet/src/handleAdmin.go:214]: You didn't ask but I think the same question applies here?

File: xsuite/src/world/lsworld.test.ts [hunk 6]
```diff
@@ -625,6 +654,76 @@ test.concurrent("LSWorld.callContract", async () => {
...
+  });
+});
+
+test.concurrent("LSWorld.addEsdts", async () => {
+  using world = await LSWorld.start();
+  const wallet = await world.createWallet({
+    balance: 10n ** 18n,
+    kvs: {
+      esdts: [
+        { id: "01", amount: 10n ** 18n },
```
Comment (lcswillems) [inline xsuite/src/world/lsworld.test.ts:679]: Let's put a well formatted ID, e.g. "TOK1-123456". Same for the other ones.

File: xsuite/src/world/lsworld.test.ts [hunk 7]
```diff
@@ -625,6 +654,76 @@ test.concurrent("LSWorld.callContract", async () => {
...
+  });
+});
+
+test.concurrent("LSWorld.addMappers", async () => {
+  using world = await LSWorld.start();
+  const wallet = await world.createWallet({
+    balance: 10n ** 18n,
+    kvs: {
+      mappers: [
+        { key: "01", value: e.Buffer("02") },
```
Comment (lcswillems) [inline xsuite/src/world/lsworld.test.ts:706]: For the key, let's put a name, e.g. "key1"

For the value, let's use bigint, e.g. `e.U(2)`.

Same for the other ones.

File: xsuite/src/world/lsworld.test.ts [hunk 8]
```diff
@@ -355,6 +355,35 @@ test.concurrent("LSWorld.setAccount", async () => {
   });
 });
 
+test.concurrent("LSWorld.updateAccount", async () => {
+  using world = await LSWorld.start();
+  const { wallet, wallet2, contract } = await createAccounts(world);
```
Comment (lcswillems) [inline xsuite/src/world/lsworld.test.ts:360]: Let's directly create the account (wallet or contract) without the helper, store the state in a variable and in the final assert, show that the state is the state before with a field changed (similar to what's done in LSWallet.updateAccount).

File: xsuite/src/world/lsworld.test.ts [hunk 9]
```diff
@@ -766,6 +865,15 @@ test.concurrent("LSWallet.setAccount & LSWallet.getAccount", async () => {
   expect(after).toEqual(before);
 });
 
+test.concurrent("LSWallet.updateAccount", async () => {
+  using world = await LSWorld.start();
+  const { wallet } = await createAccounts(world);
+  const before = await wallet.getAccount();
+  await wallet.updateAccount({ balance: 10n ** 17n });
+  const after = await wallet.getAccount();
+  expect(after).toEqual({ ...before, balance: 10n ** 17n });
```
Comment (lcswillems) [inline xsuite/src/world/lsworld.test.ts:874]: The only thing is that we don't know what "before" is. Let's directly create the wallet without the helper.

File: xsuite/src/world/lsworld.test.ts (hunk 6 above)
Comment (lcswillems) [reply xsuite/src/world/lsworld.test.ts:679]: Applies to all the other tests of the kind.

File: xsuite/src/world/lsworld.test.ts (hunk 7 above)
Comment (lcswillems) [reply xsuite/src/world/lsworld.test.ts:706]: Applies to all the other tests of the kind.

File: xsuite/src/world/lsworld.test.ts [hunk 10]
```diff
@@ -1083,6 +1234,15 @@ test.concurrent("LSContract.setAccount & LSContract.getAccount", async () => {
   expect(after).toEqual(before);
 });
 
+test.concurrent("LSContract.updateAccount", async () => {
+  using world = await LSWorld.start();
+  const { contract } = await createAccounts(world);
+  const before = await contract.getAccount();
+  await contract.updateAccount({ balance: 10n ** 17n });
+  const after = await contract.getAccount();
+  expect(after).toEqual({ ...before, balance: 10n ** 17n });
```
Comment (lcswillems) [inline xsuite/src/world/lsworld.test.ts:1243]: Same remark than for the LSWallet

**Review by lcswillems — CHANGES_REQUESTED** [review]

File: xsuite/src/world/fsworld.ts (hunk 1 above)
Comment (nvergez) [reply xsuite/src/world/fsworld.ts:206]: #219 for the eslint rule

**Review by lcswillems — APPROVED** [review]

**Pull request merged**